akka-cluster-sharding/src/main/scala/akka/cluster/sharding/ClusterSharding.scala

```diff
@@ -459,7 +459,8 @@ private[akka] class ClusterShardingGuardian extends Actor {
         val encName = URLEncoder.encode(typeName, ByteString.UTF_8)
         val cName = coordinatorSingletonManagerName(encName)
         val cPath = coordinatorPath(encName)
-        val shardRegion = context.child(encName).getOrElse {
+        val shardRegion = context.child(cName).flatMap(_ ⇒ context.child(encName)).getOrElse {
```

Comment:
It would be great if you can try different naming of the proxy. It also has to be done in ClusterSharding extension, because they share the same ConcurrentHashMap there.